From 25b408bcea0349c7909fae3e4fbbdd337ceeca3a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:27 -0400
Subject: [tools] perf/trace: Don't print zeroed args

Message-id: <1408462094-14194-99-git-send-email-jolsa@redhat.com>
Patchwork-id: 88067
O-Subject: [PATCH RHEL7.1 BZ1131394 098/325] perf trace: Don't print zeroed args
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 22ae5cf1c9b4e27a3efe96d8fa6d038560a2cf23
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Sep 12 11:27:34 2013 -0300

description
===========
This way we make the output more compact.

If somebody complain (and provide a sane reason why we would like to see
zeroes) we can make it an optional, ~/.perfconfig configurable knob.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 38782a7..04b632d 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -818,6 +818,9 @@ static size_t syscall__scnprintf_args(struct syscall *sc, char *bf, size_t size,
    if (arg.mask & bit)
     continue;
 
+   if (args[arg.idx] == 0)
+    continue;
+
    printed += scnprintf(bf + printed, size - printed,
           "%s%s: ", printed ? ", " : "", field->name);
    if (sc->arg_scnprintf && sc->arg_scnprintf[arg.idx]) {
-- 
1.7.1